scvi/inference/posterior.py

```diff
@@ -381,7 +384,8 @@ def scale_sampler(
                 np.arange(len(self.gene_dataset))[selection], n_samples
             )
             self.update_sampler_indices(idx=idx)
-            px_scales.append(self.get_sample_scale(transform_batch=batch_idx))
+            px_scales.append(self.get_sample_scale(transform_batch=batch_idx, **kwargs))
+            batch_idx = batch_idx if batch_idx is not None else np.nan
```

Comment:
np.nan? That does not look very good

Reply:
This is due to the line directly below

` batch_ids.append(np.ones((px_scales[-1].shape[0])) * batch_idx)`

This actually fixes the bug, because you can't multiply by None.

Basically, this is for the `use_observed_batches` mode. I agree this is not straightforward. Somehow we should consider a differential expression module that cleans up a lot of the code base.

Reply:
hum ok, I guess it's just unsual to assign something to NaN. Can we do something more readable?

Reply:
I propose we leave this now, and I will make an issue to fix at a later time